Description: Use libtool to build shared libraries.
 Upstream only builds static libraries.  We also install the headers
 in /usr/include/topcom instead of /usr/include.
Author: Doug Torrance <dtorrance@debian.org>
Forwarded: not-needed
Last-Update: 2022-10-03

--- a/configure.ac
+++ b/configure.ac
@@ -10,7 +10,7 @@
 
 dnl set some compilation switches consistently:
 AC_SUBST([AM_CPPFLAGS], ["-DSTL_SYMMETRIES -DSTL_CONTAINERS -DSTL_FLIPS -DSTL_CHIROTOPE -DGMPRATIONAL -DPERMLIB_DOMAIN_INT -DUSE_LEXORDER]")
-AC_SUBST([AM_CXXFLAGS], [" -fvisibility=hidden -fvisibility-inlines-hidden -pthread -std=c++17"])
+AC_SUBST([AM_CXXFLAGS], [" -pthread -std=c++17"])
 
 dnl backup the user variable LDFLAGS:
 dnl AC_SUBST([LDFLAGS_backup], [${LDFLAGS}])
@@ -22,7 +22,8 @@
 AC_PROG_CC
 AC_PROG_INSTALL
 AC_PROG_CXX
-AC_PROG_RANLIB
+
+LT_INIT
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
--- a/lib-src/Makefile.am
+++ b/lib-src/Makefile.am
@@ -1,6 +1,8 @@
-lib_LIBRARIES       = libTOPCOM.a
+lib_LTLIBRARIES       = libTOPCOM.la
 
-libTOPCOM_a_SOURCES = \
+libTOPCOM_la_LIBADD = -lcddgmp -lgmp -lgmpxx
+
+libTOPCOM_la_SOURCES = \
 Admissibles.cc \
 CheckTriang.cc \
 Circuits.cc \
@@ -54,7 +56,7 @@
 VirtualChiro.cc \
 Volumes.cc
 
-include_HEADERS = \
+pkginclude_HEADERS = \
 Admissibles.hh \
 Array.hh \
 CheckTriang.hh \
--- a/wrap-gmp-gmpxx/Makefile.am
+++ b/wrap-gmp-gmpxx/Makefile.am
@@ -1,4 +1,4 @@
-include_HEADERS = \
+pkginclude_HEADERS = \
 Integer.h \
 Rational.h
 
--- a/lib-src-reg/Makefile.am
+++ b/lib-src-reg/Makefile.am
@@ -1,6 +1,8 @@
-lib_LIBRARIES = libCHECKREG.a
+lib_LTLIBRARIES = libCHECKREG.la
 
-libCHECKREG_a_SOURCES = \
+libCHECKREG_la_LIBADD = -lcddgmp -lqsopt_ex
+
+libCHECKREG_la_SOURCES = \
 QSOinterface.cc \
 SPXinterface.cc \
 LPinterface.cc \
--- a/src-reg/Makefile.am
+++ b/src-reg/Makefile.am
@@ -2,8 +2,8 @@
 
 topcom_checkregularity_SOURCES = checkregularity.cc
 
-LDADD           = ../lib-src/libTOPCOM.a \
-                  ../lib-src-reg/libCHECKREG.a \
+LDADD           = ../lib-src/libTOPCOM.la \
+                  ../lib-src-reg/libCHECKREG.la \
                   -lcddgmp
 if USE_QSOPTEX
 LDADD          += -lqsopt_ex
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -121,8 +121,8 @@
 topcom_santos_dim4_triang_SOURCES         = santos_dim4_triang.cc
 topcom_santos_22_triang_SOURCES           = santos_22_triang.cc
 
-LDADD           = ../lib-src/libTOPCOM.a \
-                  ../lib-src-reg/libCHECKREG.a
+LDADD           = ../lib-src/libTOPCOM.la \
+                  ../lib-src-reg/libCHECKREG.la
 if USE_QSOPTEX
 LDADD          += -lqsopt_ex
 endif
